pkg/kritis/cryptolib/pgp_test.go

```diff
@@ -136,3 +304,100 @@ func TestVerifyPgp(t *testing.T) {
 		})
 	}
 }
+
+func TestCreateAttestation_WellFormedAttestation(t *testing.T) {
+	tcs := []struct {
+		name               string
+		privateKey         []byte
+		expectNewSignerErr bool
+	}{
+		{
+			name:               "correct signature over payload",
+			privateKey:         []byte(gpgPrivateKey),
+			expectNewSignerErr: false,
+		},
+		{
+			name:               "invalid private key",
+			privateKey:         []byte("invalid-private-key"),
+			expectNewSignerErr: true,
+		},
+	}
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			signer, err := NewPgpSigner(tc.privateKey)
+			if tc.expectNewSignerErr {
+				if err == nil {
+					t.Fatalf("Expected error but returned none")
+				}
+			} else {
+				if err != nil {
+					t.Fatalf("Unexpected error: %v", err)
+				}
+				att, err := signer.CreateAttestation([]byte(payload))
+				if err != nil {
+					t.Fatalf("Error creating the Attestation: %v", err)
+				}
+
+				// Check attestation contents
+				if att.SerializedPayload != nil {
+					t.Errorf("SerializedPayload does not match expected: got: %v, want: %v", att.SerializedPayload, nil)
+				}
+				if att.PublicKeyID != gpgPublicKeyID {
+					t.Errorf("PublicKeyID does not match expected: got: %q, want %q", att.PublicKeyID, gpgPublicKeyID)
+				}
+			}
+		})
+	}
+}
+
+// Contents of generated signatures depend on external factors, including time
+// and date. A signature is valid if it can be used to verify the Attestation.
+func TestCreateAttestation_ValidSignature(t *testing.T) {
+	tcs := []struct {
+		name            string
+		privateKey      []byte
+		expectVerifyErr bool
+	}{
+		{
+			name:            "valid signature over the payload",
+			privateKey:      []byte(gpgPrivateKey),
+			expectVerifyErr: false,
+		},
+		{
+			name:            "wrong private key",
```

Comment:
This test is a little misleading and I'm not sure its necessary. The signature thats created in this case is valid but the private key doesn't match the public key in the verification process. I believe you cover that case in the verification test cases. I think here you should focus on the creation of the attestation and the ability to create a well-formed signature.

Reply:
+1

Reply:
Done. Removed this test case